spec/nylas/model_spec.rb

```diff
@@ -42,6 +42,14 @@ def example_instance_hash
     end
   end
 
+  describe "#reload" do
+    it "does not explode if using an invalid attribute" do
+      allow(api).to receive(:execute).and_return(fake_attribute: "not real")
+      instance = FullModel.from_json('{ "id": 1234 }', api: api)
+      expect { instance.reload }.not_to raise_error
```

Comment:
I think it would be nice for this to include assertions that `id` is set on the model but `fake_attribute` is not. I think it would also be nice to include a test where an attribute does get set successfully on reload.

Reply:
Yea, I can add more fields to the model to show that the other stuff gets set.